web/views.py

```diff
@@ -272,17 +147,178 @@ def post(self, request, *args, **kwargs):
         if form.is_valid():
             form.save()
             if form_name == 'form2':
+                # If updating password, need to reauthenticate
                 update_session_auth_hash(self.request, form.user)
                 return HttpResponseRedirect(self.get_success_url())
             return super().form_valid(form)
         else:
             return self.form_invalid(**{form_name: form})
 
+
+class ChildrenListView(LoginRequiredMixin, generic.CreateView):
+    """
+    Allows user to view a list of current children and add children
+    """
+    template_name = 'web/children-list.html'
+    model = Child
+    form_class = forms.ChildForm
+
+    def get_context_data(self, **kwargs):
+        """
+        Add children that have not been deleted that belong to the current user
+        to the context_dict.  Also add info to hide the Add Child form on page load.
+        """
+        context = super().get_context_data(**kwargs)
+        children = Child.objects.filter(deleted = False, user=self.request.user)
+        context["objects"] = children
+        context["form_hidden"] = kwargs.get('form_hidden', True)
+        return context
+
+    def form_invalid(self, form):
+        """
+        If form invalid, add child form needs to be open when page reloads.
+        """
+        return self.render_to_response(self.get_context_data(form=form, form_hidden=False))
+
+    def form_valid(self, form):
+        """
+        Add the current user to the child before saving the child.
+        """
+        user = self.request.user
+        form.instance.user = user
+        return super().form_valid(form)
+
+    def get_success_url(self):
+        return reverse('web:children-list')
+
+
+class ChildUpdateView(LoginRequiredMixin, generic.UpdateView):
+    """
+    Allows user to update or delete a child.
+    """
+    template_name = 'web/child-update.html'
+    model = Child
+    form_class = forms.ChildForm
+
+    def get_success_url(self):
+        return reverse('web:children-list')
+
+    def get_object(self, queryset=None):
+        '''
+        Returns the object the view is displaying.
+        By default this requires `self.queryset` and a `pk` or `slug` argument
+        in the URLconf, but subclasses can override this to return any object.
+        '''
+        if queryset is None:
+            queryset = self.get_queryset()
+        # ChildUpdate View needs to be called with slug or pk - uuid in URLconf
+        # instead so use this to lookup child
+        uuid = self.kwargs.get('uuid')
+
+        if uuid is not None:
+            queryset = queryset.filter(uuid=uuid)
```

Comment:
[This](https://docs.djangoproject.com/en/1.11/topics/http/shortcuts/#get-object-or-404) can cut out some boilerplate.